pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java

```diff
@@ -294,14 +293,19 @@ private static void validateIndexingConfig(@Nullable IndexingConfig indexingConf
         noDictionaryColumnsSet.add(columnName);
       }
     }
+    Set<String> bloomFilterColumns = new HashSet<>();
     if (indexingConfig.getBloomFilterColumns() != null) {
-      for (String columnName : indexingConfig.getBloomFilterColumns()) {
-        if (noDictionaryColumnsSet.contains(columnName)) {
-          throw new IllegalStateException(
-              "Cannot create a Bloom Filter on column " + columnName + " specified in the noDictionaryColumns config");
-        }
-        columnNameToConfigMap.put(columnName, "Bloom Filter Config");
+      bloomFilterColumns.addAll(indexingConfig.getBloomFilterColumns());
+    }
+    if (indexingConfig.getBloomFilterConfigs() != null) {
+      bloomFilterColumns.addAll(indexingConfig.getBloomFilterConfigs().keySet());
+    }
+    for (String bloomFilterColumn : bloomFilterColumns) {
+      if (noDictionaryColumnsSet.contains(bloomFilterColumn)) {
+        throw new IllegalStateException("Cannot create a Bloom Filter on column " + bloomFilterColumn
```

Comment:
Not related to this topic, but somehow I feel we should allow creating bloomfilter for no-dictionary column ?

Reply:
True. Added a TODO in the `BloomFilterHandler` to support raw index